src/arwave.py

```diff
@@ -1,122 +1,219 @@
+    #****column index 0 is location name maybe? observatory name?
+ #   latitude = slf._spect['mosaic']['latitude']
+ #   longitude = slf._spect['mosaic']['longitude']
+ #   sky_files = Table.read(~/PYPIT/data/sky_files/sky_spectra_table, format='ascii')
+ #   archive_wave =
+ #   archive_flux =
+    msgs.work("Consider doing 2 passes in flexure as in LowRedux")
+
+    # Read archived sky
+    root = slf._argflag['run']['pypitdir']
+    #root = '/Users/xavier/local/Python/PYPIT'
+    #skyspec_fil = 'sky_LRISr_600_7500_5460_7950.fits'
+    if slf._argflag['reduce']['flexure']['archive_spec'] is None:
+        # Red or blue?
+        if slf._argflag['run']['spectrograph'] in ['kast_blue', 'lris_blue']:
+            skyspec_fil = 'sky_kastb_600.fits'
+        else:
+            skyspec_fil = 'paranal_sky.fits'
     else:
-        arx_sky=arx_sky.rebin(keep_wave*u.AA)
-        obj_sky=obj_sky.rebin(keep_wave*u.AA)
-
-    #deal with bad pixels
-
-    #deal with underlying continuum
-
-
-    #rebin the spectra onto the same pixel-wavelength scale
-#    n_arx=len(arx_sky.dispersion)
-#    n_obj=len(obj_sky.dispersion)
-#
-#    if n_arx > n_obj:
-#        obj_sky=obj_sky.rebin(arx_sky.dispersion)
-#        print "n_archive > n_obj, rebinning object to", len(obj_sky.dispersion)
-#
-#    elif n_arx < n_obj:
-#        print arx_sky.dispersion
-#        arx_sky=arx_sky.rebin(obj_sky.dispersion)
-#        print "n_obj > n_archive, rebinning archive to", len(arx_sky.dispersion)
-
-    #else:
-        # tell code to continue when archived and object wavelengths are the
-        # same length? do I need to?
-
-
-    #cross correlate the two spectra, which are now on the same wavelength
-    #range and have the same array sizes
-    corr=np.correlate(arx_sky.flux,obj_sky.flux,"same")
-    plt.plot(corr)
-
-    #create an array around the max of the correlation function
-    max_corr=np.argmax(corr)
-    subpix_grid=np.linspace(max_corr-4,max_corr+4,9.)
-
-    #fit a gaussian to where the correlation function peaks
-    fit,other=arutils.gauss_lsqfit(subpix_grid,corr[subpix_grid.astype(np.int)],max_corr)
-
-    #calculate the shift and apply it to the wavelength
-    shift=fit[1]-0.
-    #model=fit[0]*(np.exp((-(subpix_grid-fit[1])**2)/(2*fit[2]**2)))
-    #plt.plot(subpix_grid,model)
-
-    finer_subpix_grid=np.linspace(max_corr-4,max_corr+4,90.)
-    model=fit[0]*(np.exp((-(finer_subpix_grid-fit[1])**2)/(2*fit[2]**2)))
-    plt.plot(finer_subpix_grid,model)
-    #plt.show()
-
-    return
-
-#flexure(object_wave2,object_flux)
+        skyspec_fil = slf._argflag['reduce']['flexure']['archive_spec']
+    msgs.info("Using {:s} file for Sky spectrum".format(skyspec_fil))
+    hdu = fits.open(root+'/data/sky_spec/'+skyspec_fil)
+    archive_wave = hdu[0].data
+    archive_flux = hdu[1].data
+    arx_sky = xspectrum1d.XSpectrum1D.from_tuple((archive_wave, archive_flux))
+
+    # Loop on objects
+    flex_dict = dict(polyfit=[], roots=[], shift=[], subpix=[], corr=[],
```

Comment:
No longer need "roots" here. (Note, there's also a few instances of "roots" in the ipython notebook `Flexure_Tests.ipynb`)